xcdat/dataset.py

```diff
@@ -203,66 +200,49 @@ def open_mfdataset(
-    # https://github.com/pydata/xarray/issues/2436). The workaround is to store
-    # the time encoding from the first dataset as a variable, and add the time
-    # encoding back to final merged dataset.
-    time_encoding = None
-
-    if decode_times:
-        time_encoding = _keep_time_encoding(paths)
-
-        cf_compliant_time: Optional[bool] = _has_cf_compliant_time(paths)
-        # xCDAT attempts to decode non-CF compliant time coordinates using the
-        # preprocess keyword arg with `xr.open_mfdataset()`.
-        if cf_compliant_time is False:
-            decode_times = False
-            preprocess = partial(_preprocess_non_cf_dataset, callable=preprocess)
+    preprocess = partial(_preprocess, decode_times=decode_times, callable=preprocess)
 
     ds = xr.open_mfdataset(
         paths,
-        decode_times=decode_times,
+        decode_times=False,
         data_vars=data_vars,
         preprocess=preprocess,
         **kwargs,  # type: ignore
     )
-    ds = _postprocess_dataset(ds, data_var, center_times, add_bounds, lon_orient)
 
-    if time_encoding is not None:
-        time_dim = get_axis_dim(ds, "T")
-        ds[time_dim].encoding = time_encoding
-        # Update "original_shape" to reflect the final time coordinates shape.
-        ds[time_dim].encoding["original_shape"] = ds[time_dim].shape
+    ds = _postprocess_dataset(ds, data_var, center_times, add_bounds, lon_orient)
 
     return ds
 
 
-def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
-    """Decodes time coordinates and time bounds with non-CF compliant units.
+def decode_time(dataset: xr.Dataset) -> xr.Dataset:  # noqa: C901
+    """Decodes CF and non-CF time coordinates and time bounds using ``cftime``.
 
     By default, ``xarray`` only supports decoding time with CF compliant units
-    [3]_. This function enables decoding time with non-CF compliant units.
+    [3]_. This function enables also decoding time with non-CF compliant units.
+    It skips decoding time coordinates that have already been decoded as
+    "datetime64[ns]" or `cftime.datetime`.
 
-    The time coordinates must have a "calendar" attribute set to a CF calendar
-    type supported by ``cftime`` ("noleap", "360_day", "365_day", "366_day",
-    "gregorian", "proleptic_gregorian", "julian", "all_leap", or "standard")
-    and a "units" attribute set to a supported format ("months since ..." or
-    "years since ...").
+    For time coordinates to be decodable, they must have a "calendar" attribute
+    set to a CF calendar type supported by ``cftime``. CF calendar types
+    include "noleap", "360_day", "365_day", "366_day", "gregorian",
+    "proleptic_gregorian", "julian", "all_leap", or "standard". They must also
+    have a "units" attribute set to a format supported by xcdat ("months since
+    ..." or "years since ...").
 
-    The logic for this function:
+    How this function decodes time coordinates:
```

Comment:
Update this to include info on using `decode_cf_datetime()` for CF compliant units.